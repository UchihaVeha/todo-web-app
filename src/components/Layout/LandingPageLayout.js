// @flow
import * as React from 'react';
import { compose, pure } from 'recompose';
import Grid from 'material-ui/Grid';
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
  root: {
    display: 'flex',
    minHeight: '100vh',
    flexDirection: 'column',
    [theme.breakpoints.up('md')]: {
      maxWidth: 960,
      margin: '0 auto'
    }
  },
  content: {
    flexGrow: 1,
    textAlign: 'center'
  }
});

type Props = {
  children: React.ComponentType<*>,
  classes: Return<typeof styles>
};

const LandingPageLayout = ({ children, classes }: Props) => (
  <div className={classes.root}>
    <Grid container direction="column">
      <Grid item lg={12}>
        <h1>Header</h1>
      </Grid>
      <Grid item lg={12} className={classes.content}>
        {children}
      </Grid>
      <Grid item lg={12}>
        <h1>Footer</h1>
      </Grid>
    </Grid>
  </div>
);

export default compose(pure, withStyles(styles, { name: 'LandingPageLayout' }))(
  LandingPageLayout
);
