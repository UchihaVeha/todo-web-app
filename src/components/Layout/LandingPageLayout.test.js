import React from 'react';
import { shallow } from 'enzyme';
import LandingPageLayout from './LandingPageLayout';

describe('Layout/LandingPageLayout component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <LandingPageLayout>
        <span>Content</span>
      </LandingPageLayout>
    );
  });

  it('should render withStyles', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render component', () => {
    expect(wrapper.dive().dive()).toMatchSnapshot();
  });
});
