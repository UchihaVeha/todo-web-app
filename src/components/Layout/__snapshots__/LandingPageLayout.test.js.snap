// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/LandingPageLayout component should render component 1`] = `
<div
  className="LandingPageLayout-root-1"
>
  <withStyles(GridWrapper)
    className="LandingPageLayout-fullHeight-2"
    container={true}
    direction="column"
  >
    <withStyles(GridWrapper)
      item={true}
      lg={12}
    />
    <withStyles(GridWrapper)
      className="LandingPageLayout-fullHeight-2"
      item={true}
      lg={12}
    >
      <span>
        Content
      </span>
    </withStyles(GridWrapper)>
    <withStyles(GridWrapper)
      item={true}
      lg={12}
    />
  </withStyles(GridWrapper)>
</div>
`;

exports[`Layout/LandingPageLayout component should render withStyles 1`] = `
<LandingPageLayout
  classes={
    Object {
      "fullHeight": "LandingPageLayout-fullHeight-2",
      "root": "LandingPageLayout-root-1",
    }
  }
>
  <span>
    Content
  </span>
</LandingPageLayout>
`;
